### DISCUSSION TITLE
Penalty search method

### DISCUSSION THREAD

**Discussion by tkanzai** [deepcharles/ruptures#251] [Q&A]
Hi!

First of all, congruatulations on Ruptures package!

I was wondering if you guys are planning on including some automatic penalty method (like **CROPS**, available in R) in Ruptures. I've been struggling in finding good values for penalties and, sometimes, it all feels a little bit abstract.

Thanks in advance!

**Comment by deepcharles** [comment]
hi, thanks for your interest in ruptures.

There is no plan to include CROPS but if there is some interest, we can add it to the list of thins to implement.

How would you use it to find good values for penalties (which, I agree, is certainly a very difficult task)?

**Reply by tkanzai** [reply]
Hi, thanks for your response!

Well, I always start using some linear penalty term based on log(n), but, at the end of the day, I end up tuning this parameter somehow... Recently, I plugged an optimizer in order to provide me the changepoints trying to minimize the total number changes. That's what I'd assume CROPS would do for me in a better way. In other words, I was looking foward not to delegate this choice to the scrutiny of the user. I'm not sure if that made sense :D, but that would be my goal with an algorithm such as CROPS.

**Reply by deepcharles** [reply]
Hi, I am not sure that you can have CROPS make the choice for you. To me, it can only help you compute the segmentations for a range of penalty values. You still have to choose the value that fits your needs.

If you have some labels (e.g. manually segmented signals), you can use a supervised approach such as [this one](https://ieeexplore.ieee.org/document/8081473) ([pdf here](https://deepcharles.github.io/files/TOV-ICASSP-19.pdf)).
